msal_extensions/cache_lock.py

```diff
@@ -25,7 +28,24 @@ def __init__(self, lockfile_path):
             flags=portalocker.LOCK_EX | portalocker.LOCK_NB,
             **open_kwargs)
 
+    def try_to_create_lock_file(self):
+        retries_no = 100
+        retry_delay_milliseconds = 0.150
+        for i in range(retries_no):
+            try:
+                open(self._lockpath, 'x')
```

Comment:
Do we need to wrap this inside a `with ...` clause so that the open file handler would be closed before it being opened (again) later? It might now really matter, but feels the right thing to do...